sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,278 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_kmac.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+// TODO, change to use dif_flash_ctrl.h
+#include "sw/device/lib/flash_ctrl.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/keymgr.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/keymgr_binding_value.h"
+#include "sw/device/silicon_creator/lib/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "keymgr_regs.h"  // Generated.
+#include "kmac_regs.h"    // Generated.
+
+#define ASSERT_OK(expr_)                        \
+  do {                                          \
+    rom_error_t local_error_ = expr_;           \
+    if (local_error_ != kErrorOk) {             \
+      LOG_ERROR("Error at line: %d", __LINE__); \
+      return local_error_;                      \
+    }                                           \
+  } while (0)
+
+#define ASSERT_EQZ(x) CHECK((x) == 0)
```

Comment:
This is a result of using rom drivers. Not needed otherwise.

Reply:
`ASSERT_EQZ` is used in some flash function. Will clean up those flash functions to use dif in the next PR.